pkgs/tools/networking/veilid/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, stdenv
+, AppKit
+, Security
+, fetchFromGitLab
+, rustPlatform
+, protobuf
+, capnproto
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "veilid";
+  version = "0.1.7";
+
+  src = fetchFromGitLab {
+    owner = "veilid";
+    repo = pname;
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-wG8uxmohIOb8V+5gqhjM4hHG/6uHg0ehAtP2z5eoflU=";
+  };
+
+  cargoLock = {
+     lockFile = ./Cargo.lock;
+     outputHashes = {
+       "bugsalot-0.2.2" = "sha256-9zLzK22dOB7w+ejk1SfkA98z4rEzrB6mAVUpPFuDUnY=";
+     };
+   };
+
+  nativeBuildInputs = [
+    capnproto
+    protobuf
+  ];
+
+  buildInputs = lib.optionals stdenv.isDarwin [ AppKit Security ];
+
+  cargoBuildFlags = [
+    "--workspace"
+  ];
+
+  doCheck = false;
+
+  outputs = [ "out" "lib" ];
+
+  postInstall = ''
+    moveToOutput "lib" "$lib"
+  '';
+
+  meta = with lib; {
+    description = "Veilid allows anyone to build a distributed, private app";
```

Comment:
Description should not be self-referent.

Reply:
from the project homepage:

> Veilid is an open-source, peer-to-peer, mobile-first, networked application framework.

so how about

> An open-source, peer-to-peer, mobile-first, networked application framework.

?

Reply:
Fine.
Just remember to remove the `.` at the end.

Reply:
Good to know!

Reply:
Done, thanks!